### PR TITLE
Added the ability to pass an extension to the faye client when subscri…

### PR DESCRIPTION
**Pull request by jatkinson-vivial**
…bing to a topic. Thus allowing client to pass in an extension with replay values.
Issue: https://github.com/jsforce/jsforce/issues/547

### CODE REVIEW

File: lib/api/streaming.js [hunk 1]
```diff
@@ -127,12 +139,17 @@ Streaming.prototype._createClient = function(replay) {
...
   var isGeneric = channelName.indexOf('/u/') === 0;
   var clientType = isGeneric ? 'generic' : 'pushTopic';
   if (!this._fayeClients || !this._fayeClients[clientType]) {
     this._fayeClients = this._fayeClients || {};
     this._fayeClients[clientType] = this._createClient(isGeneric);
+
+    if (typeof extension !== "undefined") {
+      this._fayeClients[clientType].addExtension(extension)
+    }
+
```
Comment (stomita) [inline lib/api/streaming.js]: Implicitly modifies faye client with addExtension, which leads the following normal getFayeClient(channelName) calls to return the client instance with extension.
Should be much explicit API interface if you are going to modify the client behavior.

Comment (sfescape) [reply lib/api/streaming.js]: I have the opposite view, I think extensions should just be an added optional parameter when creating a Topic. The faye client is specific to the Topic so adding extensions to it should be treated the same as the channel name.

**Comment by sfescape** [comment]
Why not just add extensions as the last (optional) argument to Topic?

conn.streaming.topic(name, extensions)

That seems the most natural place to put this. Not sure why it hasn't been accepted yet.

**Comment by nolson-berry** [comment]
@stomita 
We just upgraded to the latest jsforce and the changes from @mars are working nicely.

**This PR can be closed.**  The _jatkinson-vivial_ github account is no longer active.

```javascript
const jsforce = require('jsforce')
...
function loginCallback (salesforceConnection) {
  const topic = `/topic/${process.env.SALESFORCE_TOPIC}`
  const replayExt = new jsforce.StreamingExtension.Replay(topic, -2)
  const fayeClient = salesforceConnection.streaming.createClient([replayExt])

  fayeClient.subscribe(topic, (salesforceEvent) => {
    console.log('Received: %j',salesforceEvent)
    return doSomething(salesforceEvent)
  })
}
...
```

**Pull request closed without merging**